models/user.modal.go

```diff
@@ -0,0 +1,83 @@
+package models
+
+import (
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+)
+
+type User struct {
+	ID        *uuid.UUID `gorm:"type:uuid;default:uuid_generate_v4();primary_key"`
+	FirstName string     `gorm:"type:varchar(100);not null"`
+	LastName  string     `gorm:"type:varchar(100);not null"`
+	Username  string     `gorm:"type:varchar(100);uniqueIndex;not null"`
+	Role      *string    `gorm:"type:varchar(50);default:'user';not null"`
+	Email     string     `gorm:"type:varchar(100);uniqueIndex;not null"`
+	Password  string     `gorm:"type:varchar(100);not null"`
+	Photo     *string    `gorm:"not null;default:'default.png'"`
+	CreatedAt *time.Time `gorm:"not null;default:now()"`
+	UpdatedAt *time.Time `gorm:"not null;default:now()"`
+}
+
+type SignUpInput struct {
+	FirstName       string `json:"first_name" validate:"required"`
+	LastName        string `json:"last_name" validate:"required"`
+	Username        string `json:"username" validate:"required"`
+	Email           string `json:"email" validate:"required"`
+	Password        string `json:"password" validate:"required,min=8"`
+	PasswordConfirm string `json:"password_confirm" validate:"required,min=8"`
+	Photo           string `json:"photo"`
```

Comment:
do we really need photo while doing signup

Reply:
It's optional 
If we are not likely to take we can remove it 
what is your opinion here